gitutil.py

```diff
@@ -130,14 +130,19 @@ def branch_head(self, branch_name):
         # XXX added for backward compatibility - remove
         return self.fetch_head(ref=branch_name)
 
-
-def clone_repository(
-        to_path: str,
-        github_cfg,
-        github_repo_path: str
-    ):
-        url = url_with_credentials(github_cfg, github_repo_path)
-        git.Git().clone('--quiet', url, to_path)
+    @staticmethod
```

Comment:
this commit changes more than the commit message suggests; making a method static would result in exactly two changes:
- rm 'self' argument
- add @staticmethod decorator

I would also agree to squash this commit e.g. into the previous one, or to keep it as it is. in the latter case, please at least do not "lie" in the commit msg.

Again: also adjust users. Note that doing so will break consuming pipelines. So the change ought pbly be introduced in a backward-compatiable manner. Then a new version of cc-job-image needs to be released and rolled out. Then the usages need to be changed to the new API. then the backward-compatibility coding may be reomoved.